DataDog/apps
# chore: update react-scripts to v5

## Motivation

This updates react-scripts in the generated starter app from `@datadog/create-app` in order to resolve the `react-error-overlay` issue that was preventing hot reloading from working while developing an app after using `yarn create @datadog/app`.

**References**:
- https://github.com/facebook/create-react-app/issues/11771
- https://stackoverflow.com/a/70452191/656011

## Changes

Upgrade to `v5.0.0` of `react-scripts`

## Testing

### To See the Current Issue

1. Clear yarn cache and run `yarn create @datadog/app`
2. Run `yarn start` in generated app
3. Make change to `src/widget/index.tsx` in generated app
4. Notice that hot reload does not work

### To Test the Idea Behind This Fix

1. Delete `node_modules` and `yarn.lock`
2. Update `react-scripts` version in `package.json` to `^5.0.0`
3. Run `yarn`
4. Repeat steps 2 & 3 from the [To See the Current Issue](#to-see-the-current-issue) above
5. Notice that hot reload does work

**To Test this Fix**:

1. Checkout this branch
2. `yarn link` the `@datadog/create-app` package
3. Run `yarn create @datadog/app --commit 52f5e05d00077915e8a7ed424d4bd83ae6e1fe85`
4. Run `yarn start` in generated app
5. Make a change to `src/widget/index.tsx` in generated app
6. Notice that hot reload does work
7. Shut down the development server
8. `yarn unlink` the `@datadog/create-app` package

## Releases

Choose one:

-   [ ] No release is necessary.
        If you're only updating examples/documentation, this is likely what you want.
-   [x] All packages that need a release have a changeset
        Only  `@datadog/create-app` needs a release
